### ISSUE TITLE
Export ACL List is missing some data

### ISSUE THREAD

**Issue by btzq** [apache/cloudstack#8862]
##### ISSUE TYPE

 * Improvement Request

##### COMPONENT NAME

~~~
Improvement Request (UI, Functionality)
~~~

##### CLOUDSTACK VERSION

~~~
4.19
~~~

##### CONFIGURATION

N/A

##### OS / ENVIRONMENT

N/A

##### SUMMARY

Note: This ticket is broken down as requested from the origianl post, https://github.com/apache/cloudstack/issues/8841

**Export ACL List is missing some data**

- For example, 'Ports' are not available in the exported CSV when exported.
- There may be others that are missing from the CSV output too.
- Exporting the full list as CV makes it very convenient to study, review the rules all at one go using excel. 
- If it paired together with 'Import ACL from CSV' https://github.com/apache/cloudstack/issues/8858 , it would make ACL much easier to use all round.

##### STEPS TO REPRODUCE

N/A

~~~
N/A
~~~

##### EXPECTED RESULTS

~~~
To be able to manage ACL Rules more easily for large scale projects
~~~

##### ACTUAL RESULTS

~~~
Challenging to use ACL rules for large scale projects.
~~~

**Comment by DaanHoogland** [comment]
@btzq , with "export", do you mean the output of a tool like cmk?
This seems a UI bug; if the first rule does not have start and end port these wont be in the exported CVS.
Note that the CVS export is a UI feature per se, so no API feature other than the response containing it or not.
This can be solved in the UI or in response (return -1?). I think UI makes more sense.

**Comment by btzq** [comment]
@DaanHoogland when i meant 'Export', i meant the existing export function. Exporting via CMK is not required (For us at least). 

My thoughts are that if a ACL rule doesnt have a start and end port, the columns should still be exported anyways with just empty value. Putting as -1 instead of empty value also makes sense. 

Yes i think this is just a UI issue, the data is already there. So this is another low hanging fruit.

**Comment by shwstppr** [comment]
@btzq I tried to check the csv for a rule without any ports and it shows `undefined` against port column. Is this not correct or am I missing something?
![image](https://github.com/apache/cloudstack/assets/153340/ab3ed634-e3a5-453c-a660-71392a6496e5)

**Comment by weizhouapache** [comment]
> @shwstppr , what version of Cloudstack are you using? I am unable to export the 'Startport' and 'EndPort' in the xlsx.
> 
> Im using 4.19.0.0  
> 
> This is my screenshot:
> ![Screenshot 2024-05-23 at 9 50 17 pm](https://github.com/apache/cloudstack/assets/147712084/b6a6f1ea-afd7-4f7d-960d-e1d69e65e4b3)
> 
> And this is how i exported it:
> ![Screenshot 2024-05-23 at 9 51 17 pm](https://github.com/apache/cloudstack/assets/147712084/c5fd9181-b9b8-4075-839e-572cfb33402e)
> 
> 

Yes, this is consistent with what Daan and me found. It is a ui bug

If the first rule does not have startport and endport (for example a icmp rule), the exported rules do not have startport and endport.

**Comment by shwstppr** [comment]
@btzq cc @weizhouapache I missed the fact that it needs to be the first rule without ports. I was able to reproduce and have create a fix in #9118

**Comment by shwstppr** [comment]
Fixed with #9118
